src/components/Table.tsx

```diff
@@ -0,0 +1,112 @@
+import { Table as TanStackTable, flexRender } from "@tanstack/react-table";
+
+import { Icons } from "@us-gov-cdc/cdc-react-icons";
+
+import {
+  Table,
+  TableBody,
+  TableDataCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+  TablePaginationServerSide,
+} from "@us-gov-cdc/cdc-react";
+
+export interface PortalTableProps<TData> {
+  table: TanStackTable<TData>;
+}
+
+function PortalTable<TData>({ table }: PortalTableProps<TData>) {
+  const getColSize = (field: string) => {
+    switch (field) {
+      case "filename":
+        break;
+      case "status":
+        return "sm";
+      case "timestamp":
+        return "md";
+      case "details":
+        return "sm";
+      default:
+        return "md";
+    }
+  };
```

Comment:
Refactor this to not use a switch statement. `sm` is being returned twice which is a sign that a switch statement may not be the best use case for this problem.